source/rfc/metrics.rst

```diff
@@ -0,0 +1,83 @@
+Metrics
+=======
+
+Problem
+-------
+
+Zuul jobs are run as builds on independent, created on-demand, nodes. No information is available about the
+state changes of the nodes throughout the whole buildset run in terms of CPU, memory, storage usage
+or any other useful information related to the buildset itself for that matter. The only results coming
+out of buildsets currently are test reports, run logs and built artifacts.
+
+To have better understanding and control over the CI system and any insight on how changes in code
+impact the building and testing processes, more information about the system the buildsets are being executed on
+needs to be gathered and on the outputs of the builds themselves.
+
+Metrics
+-------
+
+This section contains proposals of metrics to gather during and from buildsets.
+
+#. General
+
+   * run duration
+
+#. Physical resources
+
+   * CPU usage by time
+   * memory usage by time
+   * storage usage by time
+   * disk IO by time
+   * network IO by time
+
+   *Note: in case of storage usage, might need to consider only specific mount points*
+
+#. Test jobs specific metrics
+
+   * number of tests run for a specific buildset - might be useful in enforcing that more tests are added
+   * duration of each test case execution
+   * run log size
+
+#. Packaging jobs specific metrics
+
+   * size of created packages
+
+#. Container builder jobs specific metrics
+
+   * size of created images
+
+Technical requirements
```

Comment:
Explicitly add information that we would like to store two types of metrics: time-series and discrete values. Both of them are useful separately, i.e. we can benefit from setting up a discrete metrics gathering mechanism without the time-series one and vice-versa.

Reply:
Yes, single value per build, e.g. a contrail-build-packages build can return a single value for the disk-size of built package files.

Reply:
Added.